### ISSUE TITLE
Change _check_quantity() and _check_relativistic() to decorators

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#55]
When working on `parameters.py`, I created two temporary methods in `utils/checks.py`: `_check_quantity` and `_check_relativistic`.  `_check_quantity` does a lot of checking to make sure that a Quantity has the right units or gets the right units attached to it and has valid values.  `_check_relativistic` makes sure that the output of a function is not relativistic by raising warnings if the velocity is greater than 10% of the speed of light.  These methods would be much more elegant as decorators so that they can be called like:
```Python
from astropy import units

@check_relativistic
@check_quantity
def ion_sound_speed(whatever: units.T, ... ):
    ...
    return c_s.to(units.m/units.s)
```
Your mission, if you choose to accept it, is to change `_check_quantity` and `_check_relativistic` to decorators.  After that, the next step would be to modify `parameters.py` to use the decorators instead of the original methods.  Thank you in advance!

**Comment by namurphy** [comment]
Other possibilities would be to make a pull request to Astropy to extend the `astropy.units.quantity_input` decorator to include the functionality that we need, and to create a new Astropy decorator to check for relativistic velocities in outputs.

**Comment by GrahamGoudeau** [comment]
I've been taking a look at this. One question I had: in `physics/parameters.py`, there are two methods (`electron_gyroradius` and `Alfven_speed`) that do computation before the `_check_quantity` call. How were you imagining this working with a decorator?  I'm not a big fan of that pattern of physically swapping the contents of two of the parameters.

**Comment by SolarDrew** [comment]
Closed by #65
